### PR TITLE
fix: fixed failed to handle: base_dir() must be absolute

### LINKED ISSUE #54: Error: failed to handle: base_dir() must be absolute. Please ensure that `jsc.baseUrl` is specified correctly.
Getting the following issue when I follow the README steps. Cloned https://github.com/payloadcms/next-payload-demo and ran as is and seeing the same. Any ideas?

```
thread '<unnamed>' panicked at 'base_dir() must be absolute. Please ensure that `jsc.baseUrl` is specified correctly.', C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\swc_ecma_transforms_module-0.173.28\src\path.rs:104:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
C:\Users\Chris\code-coyotes\node_modules\@swc\core\index.js:241
            return bindings.transformSync(isModule ? JSON.stringify(src) : src, isModule, toBuffer(newOptions));
                            ^

Error: failed to handle: base_dir() must be absolute. Please ensure that `jsc.baseUrl` is specified correctly.
    at Compiler.transformSync (C:\Users\Chris\code-coyotes\node_modules\@swc\core\index.js:241:29)
    at Object.transformSync (C:\Users\Chris\code-coyotes\node_modules\@swc\core\index.js:348:21)
    at compile (C:\Users\Chris\code-coyotes\node_modules\@swc\register\lib\node.js:76:22)
    at compileHook (C:\Users\Chris\code-coyotes\node_modules\@swc\register\lib\node.js:91:16)
    at Module._compile (C:\Users\Chris\code-coyotes\node_modules\pirates\lib\index.js:113:29)
    at Module._extensions..js (node:internal/modules/cjs/loader:1166:10)
    at Object.newLoader [as .ts] (C:\Users\Chris\code-coyotes\node_modules\pirates\lib\index.js:121:7)
    at Module.load (node:internal/modules/cjs/loader:988:32)
    at Module._load (node:internal/modules/cjs/loader:834:12)
    at Module.require (node:internal/modules/cjs/loader:1012:19) {
  code: 'GenericFailure'
}

Node.js v18.4.0
```

### PR DESCRIPTION
**Pull request by walid-mos**
Error: failed to handle: base_dir() must be absolute. Please ensure that jsc.baseUrl is specified correctly.

fix #54

### CODE REVIEW

**Review by JarrodMFlesch — CHANGES_REQUESTED** [review]

File: loadPayloadConfig.js [hunk 1]
```diff
@@ -1,43 +1,46 @@
...
-    sourceMaps: 'inline',
+    sourceMaps: "inline",
     jsc: {
       parser: {
-        syntax: 'typescript',
+        syntax: "typescript",
         tsx: true,
       },
       paths: undefined,
       baseUrl: undefined,
```
[the remark this answers is not in the document]
Comment (walid-mos) [reply loadPayloadConfig.js:17]: Hello, i doubt that we can reliably get __dirname from user's projects, i think for example if payload is not initialized in root folder for example. 

But i will try with my configuration, we'll see.

File: loadPayloadConfig.js [hunk 2]
```diff
@@ -1,43 +1,46 @@
...
-  
+
   if (tsConfig?.config?.compilerOptions?.paths) {
     swcOptions.jsc.paths = tsConfig?.config?.compilerOptions?.paths;
-  
+
     if (tsConfig?.config?.compilerOptions?.baseUrl) {
-      swcOptions.jsc.baseUrl = tsConfig?.config?.compilerOptions?.baseUrl;
+      swcOptions.jsc.baseUrl = path.resolve(
+        tsConfig?.config?.compilerOptions?.baseUrl
```
[the remark this answers is not in the document]
Comment (walid-mos) [reply loadPayloadConfig.js:33]: Yes true i will correct this

File: loadPayloadConfig.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (JarrodMFlesch) [reply loadPayloadConfig.js:17]: That may be true, but I think that is what was happening previously before @swc/core made the change in 1.3.76.

Keep baseUrl set to `undefined` and try to build your project without defining a baseUrl in your tsconfig.

[the remark this answers is not in the document]
Comment (walid-mos) [reply loadPayloadConfig.js:17]: So, i tried in my config, and had no problem, it seems to work well, i am using a turborepo monorepo and src/payload dir created by the package, and works nicely.

[the remark this answers is not in the document]
Comment (JarrodMFlesch) [reply loadPayloadConfig.js:17]: I boot it up locally and it breaks if I leave it undefined and do not define a baseUrl in the tsconfig.

Might want to test it out in a more isolated repo.

**Pull request merged**